### ISSUE TITLE
Atomic operations for PowerPC

### ISSUE THREAD

**Issue by barracuda156** [nilsvanvelzen/mac_ppc_openjdk8u60#1]
I have noticed that you implemented a missing atomic operation from `gcc-4.x` specs. This may be relevant: https://github.com/classilla/tenfourfox/blob/master/js/src/jit/osxppc/AtomicOperations-ppc.h

**Comment by nilsvanvelzen** [comment]
Interesting, thanks for the link. I'll look at it.

**Comment by barracuda156** [comment]
> Interesting, thanks for the link. I'll look at it.

I think we should switch from sync* to atomic*, see: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=63368
That will require using a modern GCC though.

**Comment by nilsvanvelzen** [comment]
That is indeed an interesting thread. However, switching to an other GCC version might open another can of worms. The choice of the compiler suite is quite picky. I have tried in the past to use other gcc versions using a Mac-port install but those versions did not behave the same as the Apple GCC-versions casing compile problems at other places in the code/build

**Comment by barracuda156** [comment]
@nilsvanvelzen Yes, that is the case indeed. But recent (13+) gcc versions had several ppc-relevant bugs fixed, so I am tempted to try, as that can potentially resolve some issues.

I wonder why I get bus error when the code is supposed to work. Maybe it is my 10a190 SDK issues, though I also think we should change some fixes a bit. I will try rebuilding in a while.

(I can actually try building this on proper 10.6.8 on powerpc, that should resolve headers issues and possibly breakages, but unfortunately we have DNS and DHCP broken on 10.6.8, so it won’t be of great utility there.)
